File: FakeBet.API/FakeBet.API/Models/Bet.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FakeBet.API.Models
{
    public class Bet
    {
        [Key] public ulong Id { get; set; }

        // FK 
        public string MatchId { get; set; }

        public Match Match { get; set; }

        // FK
        public string UserId { get; set; }

        public User User { get; set; }

        public int BetOnTeamA { get; set; }

        public int BetOnTeamB { get; set; }
    }
}

File: FakeBet.Site/src/app/models/bet.ts
export class Bet {
  betId: number;
  matchId: string;
  userId: string;
  betOnTeamA: number;
  betOnTeamB: number;


  constructor(matchId: string, userId: string, betOnTeamA: number, betOnTeamB: number) {
    this.matchId = matchId;
    this.userId = userId;

    if (!betOnTeamA) {
      this.betOnTeamA = 0;
    } else {
      this.betOnTeamA = betOnTeamA;
    }

    if (!betOnTeamB) {
      this.betOnTeamB = 0;
    } else {
      this.betOnTeamB = betOnTeamB;
    }
  }
}
